File: android/src/main/java/com/jvanila/flutter/shopify/usecases/DeleteCustomerAddressUseCase.java
package com.jvanila.flutter.shopify.usecases;

import com.jvanila.flutter.pluginarch.PluginContext;
import com.shopapp.gateway.ApiCallback;
import com.shopapp.gateway.entity.Error;
import com.shopapp.shopify.api.ShopifyApi;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import kotlin.Unit;

public class DeleteCustomerAddressUseCase extends ShopifyCallUseCase {

    private static final String ARG_ADDRESS_ID = "addressId";

    public DeleteCustomerAddressUseCase(PluginContext<ShopifyApi> pluginContext) {
        super(pluginContext);
    }

    @Override
    protected void call(MethodCall input, final MethodChannel.Result result) {
        String addressId = input.argument(ARG_ADDRESS_ID);
        mPluginContext.api.instance.deleteCustomerAddress(addressId, new ApiCallback<Unit>() {
            @Override
            public void onResult(Unit unit) {
                result.success(true);
            }

            @Override
            public void onFailure(Error error) {
                System.out.println("onFailure -- " + error);
                result.error("DeleteCustomerAddressUseCase", error.getMessage(), error);
            }
        });
    }
}

File: ios/Classes/UseCases/DeleteCustomerAddressUseCase.swift
//
//  DeleteCustomerAddressUseCase.swift
//  shopify
//
//  Created by Joseph jose on 10/09/19.
//

import UIKit
import ShopApp_Shopify
import ShopApp_Gateway
import Flutter

class DeleteCustomerAddressUseCase: UseCase {

    static let ARG_ADDRESS_ID = "addressId"
    
    override public init(_ context: PluginContext) {
        super.init(context)
    }
    
    override public func trigger(with methodCall: FlutterMethodCall, result: @escaping (Any?) -> Void) {
        
        if let args = methodCall.arguments as? [String:String] {
            if let addressId = args[DeleteCustomerAddressUseCase.ARG_ADDRESS_ID] {
                
                (mContext.api.instance as! ShopifyAPI).deleteCustomerAddress(with: addressId) { (success, error) in
                    if error != nil {
                        
                        self.createError(withCase: "DeleteCustomerAddressUseCase", message: error!.errorMessage, error: error!, on: result)
                    }else {
                        
                        result(success)
                    }
                }
            }
        }
    }
}
